### ISSUE TITLE
Spinners on long running `ActionMenuItems` are not visible as dropdown is toggled on click

### ISSUE THREAD

**Issue by jams2** [wagtail/wagtail#9091]
### Issue Summary

When adding a custom `ActionMenuItem` to the page action menu, that makes use of the "spinner on click" functionality as seen on the "Save Draft" button, it is not possible to observe the spinner as the menu is closed when clicking on the button. Creating long running `ActionMenuItems` is not documented, but this also affects the Publish button (which, [in its template](https://github.com/wagtail/wagtail/blob/main/wagtail/admin/templates/wagtailadmin/pages/action_menu/publish.html), specifies the attributes for a long-running button) - although in the common case it is probably not noticeable.

### Steps to Reproduce

The easiest way I've found to reproduce this is to:
1. insert a breakpoint at [line 254 of admin/core.js](https://github.com/wagtail/wagtail/blob/main/client/src/entrypoints/admin/core.js#L254);
2. open a page for editing in admin;
3. click the "Publish" button; and
4. step through the following lines, observing that the dropdown menu is toggled.

This was raised by a client, who are trying to implement a custom long-running button. I've tested against main, client is on 3.0.1.

### Technical details

-   Python version: Python 3.8.12
-   Django version: 4.0.5
-   Wagtail version: tested against current HEAD https://github.com/wagtail/wagtail/commit/754fbb8d311fa4528fea6f1698051c5ff8757c16
-   Browser version: Firefox 103

**Comment by lb-** [comment]
Thanks @jams2  - I am not sure that this is supported right now and I do not think it really needs to be.

Are you expecting that the drop-down menu should stay open when it contains a longrunning button?

I think the goal of this kind of button is to help the user understand that something may take a long time but also to ensure they do not click it twice accidentally. In the case of things inside a dropdown menu, the additional click (if it can be done) is still blocked.

Also - the functionality works fine (assuming you have enough time to open the dropdown again).

<img width="1433" alt="Screen Shot 2022-09-07 at 3 26 30 pm" src="https://user-images.githubusercontent.com/1396140/188795771-00f1f162-4720-4137-813b-5133038c493e.png">

For now I will close this as it is unlikely to be implemented and based on this comment https://github.com/wagtail/wagtail/issues/8232#issuecomment-1218768394 - the future of the longrunning button may look different.

Please let me know if I have not understood the request correctly.

**Comment by jams2** [comment]
Thanks @lb- ,

yes, the expectation is that the dropdown would stay open if a long-running button has been clicked.

I think if the intention is to communicate to a user that "the system is responding to your input, but the result will not be instantaneous" a better outcome is that the dropdown stays open, so the user can immediately see that something is happening. If the dropdown closes, there is potential for confusion in the case of an action that takes longer than (say) 1s - "I clicked the button, but nothing appears to be happening" - as there is no other feedback in the interface.

However, if the intention of the long running button is solely to prevent duplicate clicks, I understand the position of simply closing the menu.

**Comment by Scotchester** [comment]
I agree with you, @jams2, I would prefer to have the menu stay open. As a user, I am all in favor of clear indicators that something not "instantaneous" is happening.

**Comment by lb-** [comment]
Thanks for the input all - let's re-open this for now. I've changed to an enhancement though instead of a bug.

**Comment by thibaudcolas** [comment]
Just a heads’up this will be fixed as part of #9030.

**Comment by Sharanshetty658** [comment]
has this been fixed

**Comment by laymonage** [comment]
Looks like this has been fixed in #12400 🎉
